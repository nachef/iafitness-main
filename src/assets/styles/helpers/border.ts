import { css } from 'styled-components'

export interface BorderProps {
  borderRadius?: string
  borderWidth?: string
  borderStyle?: 'solid' | 'dashed'
  borderColor?: string
}

export const BorderStyle = ({ borderWidth, borderStyle, borderRadius, borderColor }: BorderProps) => css`
  border-width: ${borderWidth ? borderWidth : '0px'};
  border-style: ${borderStyle ? borderStyle : 'solid'};
  border-radius: ${borderRadius ? borderRadius : '0px'};
  border-color: ${borderColor ? borderColor : '0px'};
`
