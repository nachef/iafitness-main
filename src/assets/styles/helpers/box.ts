import { css } from 'styled-components'

export interface BoxProps {
  width?: string
  height?: string
  display?: 'flex' | 'none' | 'block'
  flexDirection?: 'row' | 'column'
  flexJustify?: 'flex-start' | 'flex-end' | 'center' | 'space-between'
  flexAlign?: 'flex-start' | 'flex-end' | 'center'
  flexWrap?: 'wrap' | 'nowrap'
  gap?: string
}

export const BoxStyle = ({
  width,
  height,
  display,
  flexDirection,
  flexJustify,
  flexAlign,
  flexWrap,
  gap,
}: BoxProps) => css<BoxProps>`
  ${width && `width: ${width};`}
  ${height && `height: ${height};`}

  display: ${display ? display : 'block'};
  flex-direction: ${flexDirection ? flexDirection : 'column'};
  align-items: ${flexAlign ? flexAlign : 'flex-start'};
  justify-content: ${flexJustify ? flexJustify : 'flex-start'};
  flex-wrap: ${flexWrap ? flexWrap : 'nowrap'};
  gap: ${gap ? gap : '0px'};
`
