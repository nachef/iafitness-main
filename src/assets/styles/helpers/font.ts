import { css } from 'styled-components'

export interface FontProps {
  font: 'light' | 'medium' | 'semibold' | 'bold'
  size: 'xx-small' | 'x-small' | 'small' | 'regular' | 'normal' | 'medium' | 'large' | 'x-large' | 'xx-large'
  textAlign?: 'auto' | 'left' | 'right' | 'center' | 'justify'
}

export const switchFont = (value: string) => {
  switch (value) {
    case 'light':
      return 'SF Pro Display Regular'
    case 'medium':
      return 'SF Pro Display Medium'
    case 'semibold':
      return 'SF Pro Display Semibold'
    case 'bold':
      return 'SF Pro Display Bold'
  }
}

export const FontStyle = ({ font, size, textAlign }: FontProps) => css`
  font-family: ${switchFont(font)};
  font-size: ${({ theme }) => theme.utilities.sizes[size]}px;
  line-height: ${({ theme }) => theme.utilities.sizes[size] + 8}px;
  text-align: ${textAlign ? textAlign : 'left'};
`
