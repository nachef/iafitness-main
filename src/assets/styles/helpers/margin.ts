import { css } from 'styled-components'

export interface MarginProps {
  mt?: number
  mb?: number
  ml?: number
  mr?: number
  my?: number
  mx?: number
}

export const MarginStyle = ({ mt, mb, ml, mr, my, mx }: MarginProps) => css`
  margin-top: ${mt ? mt : my ? my : 0}px;
  margin-bottom: ${mb ? mb : my ? my : 0}px;
  margin-left: ${ml ? ml : mx ? mx : 0}px;
  margin-right: ${mr ? mr : mx ? mx : 0}px;
`
