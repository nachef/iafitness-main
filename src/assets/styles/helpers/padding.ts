import { css } from 'styled-components'

export interface PaddingProps {
  pt?: number
  pb?: number
  pl?: number
  pr?: number
  py?: number
  px?: number
}

export const PaddingStyle = ({ pt, pb, pl, pr, py, px }: PaddingProps) => css<PaddingProps>`
  padding-top: ${pt ? pt : py ? py : 0}px;
  padding-bottom: ${pb ? pb : py ? py : 0}px;
  padding-left: ${pl ? pl : px ? px : 0}px;
  padding-right: ${pr ? pr : px ? px : 0}px;
`
