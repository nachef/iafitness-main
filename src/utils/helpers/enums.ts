export enum Goal {
  Hipertrofia = 1,
  Emagrecimento = 2,
  DisfuncaoSexual = 3,
}
