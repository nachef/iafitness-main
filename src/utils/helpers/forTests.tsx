import React from 'react'

import { ThemeProvider } from 'styled-components'
import { render, RenderResult } from '@testing-library/react'

import { DEFAULT_THEME } from '@assets/styles/theme'

import '@testing-library/jest-dom'
import 'jest-styled-components'
import 'regenerator-runtime/runtime'

export const renderWithTheme = (children: React.ReactNode): RenderResult =>
  render(<ThemeProvider theme={DEFAULT_THEME}>{children}</ThemeProvider>)
