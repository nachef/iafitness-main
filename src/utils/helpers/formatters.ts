export function validateEmail(email: string): boolean {
  if (!email) return false // if not a string

  const pattern = /[^@ \t\r\n]+@[^@ \t\r\n]+\.[^@ \t\r\n]+/
  return pattern.test(email)
}

export function validateCPF(cpf: string): boolean {
  if (!cpf) return false // if not a string

  cpf = cpf.replace(/[^\d]+/g, '')

  if (cpf === '') return false
  else if (
    cpf.length !== 11 ||
    cpf === '00000000000' ||
    cpf === '11111111111' ||
    cpf === '22222222222' ||
    cpf === '33333333333' ||
    cpf === '44444444444' ||
    cpf === '55555555555' ||
    cpf === '66666666666' ||
    cpf === '77777777777' ||
    cpf === '88888888888' ||
    cpf === '99999999999'
  )
    return false

  let add = 0
  let i = 0
  let rev

  for (i = 0; i < 9; i += 1) add += Number(cpf.charAt(i)) * (10 - i)
  rev = 11 - (add % 11)
  if (rev === 10 || rev === 11) rev = 0
  if (rev !== Number(cpf.charAt(9))) return false

  add = 0
  for (i = 0; i < 10; i += 1) add += Number(cpf.charAt(i)) * (11 - i)
  rev = 11 - (add % 11)
  if (rev === 10 || rev === 11) rev = 0

  if (rev !== Number(cpf.charAt(10))) return false
  else return true
}

export function formatCpf(cpf: string) {
  return cpf
    .replace(/(\d{3})(\d)/, '$1.$2')
    .replace(/(\d{3})(\d)/, '$1.$2')
    .replace(/(\d{3})(\d{1,2})$/, '$1-$2')
}

export function convertToSubcurrency(amount: number, factor = 100) {
  return Math.round(amount * factor)
}

export function abbreviateWord(text: string | number): string {
  return text
    .toString()
    .replace(/segundos/g, 'seg')
    .replace(/minutos/g, 'min')
    .replace(/minuto/g, 'min')
    .replace(/seg por lado/g, 'seg p/lado')
    .replace(/Máximo de repetições/g, 'Máx')
    .replace(/Máximo de Repetições/g, 'Máx')
}

export function nameFormatterOnePart(name: string): string {
  // Converte o nome para o formato de título
  const titleCaseName = name
    .toLowerCase()
    .split(' ')
    .map(word => word.charAt(0).toUpperCase() + word.slice(1))
    .join(' ')

  // Divide o nome em partes
  const nameParts = titleCaseName.split(' ')

  // Retorna apenas os dois primeiros nomes
  return nameParts.slice(0, 1).join(' ')
}

export function nameFormatterTowParts(name: string): string {
  // Converte o nome para o formato de título
  const titleCaseName = name
    .toLowerCase()
    .split(' ')
    .map(word => word.charAt(0).toUpperCase() + word.slice(1))
    .join(' ')

  // Divide o nome em partes
  const nameParts = titleCaseName.split(' ')

  // Retorna apenas os dois primeiros nomes
  return nameParts.slice(0, 2).join(' ')
}

export const calculateBMI = (height: number, weight: number): string => {
  if (height <= 0 || weight <= 0) return ''
  const heightInMeters = height / 100
  const bmi = weight / (heightInMeters * heightInMeters)
  return bmi.toFixed(2)
}

export const FirstLetterUppercase = (text: string): string => {
  return text.charAt(0).toUpperCase() + text.slice(1)
}
