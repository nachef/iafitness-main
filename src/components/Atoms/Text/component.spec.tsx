import React from 'react'

import { renderWithTheme } from '@utils/helpers/forTests'

import { Text } from '.'

describe('Components/Atoms/Text', () => {
  it('It works', () => {
    const { container } = renderWithTheme(
      <Text color='#000000' font='bold' size='small'>
        Test text component
      </Text>,
    )

    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveTextContent('Test text component')
  })

  it('Should have text color', () => {
    const { container } = renderWithTheme(
      <Text color='#000000' font='bold' size='small'>
        Test text color
      </Text>,
    )

    expect(container.firstChild).toHaveStyleRule('color', '#000000')
  })

  it('Should have margin', () => {
    const { container } = renderWithTheme(
      <Text color='#000000' font='bold' size='small' mt={5} mb={10} ml={15} mr={20}>
        Test text color
      </Text>,
    )

    expect(container.firstChild).toHaveStyleRule('margin-top', '5px')
    expect(container.firstChild).toHaveStyleRule('margin-bottom', '10px')
    expect(container.firstChild).toHaveStyleRule('margin-left', '15px')
    expect(container.firstChild).toHaveStyleRule('margin-right', '20px')
  })

  it('Should have padding', () => {
    const { container } = renderWithTheme(
      <Text color='#000000' font='bold' size='small' pt={5} pb={10} pl={15} pr={20}>
        Test text color
      </Text>,
    )

    expect(container.firstChild).toHaveStyleRule('padding-top', '5px')
    expect(container.firstChild).toHaveStyleRule('padding-bottom', '10px')
    expect(container.firstChild).toHaveStyleRule('padding-left', '15px')
    expect(container.firstChild).toHaveStyleRule('padding-right', '20px')
  })
})
