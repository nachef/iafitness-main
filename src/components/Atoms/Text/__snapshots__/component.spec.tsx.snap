// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Atoms/Text It works 1`] = `
.c0 {
  color: #000000;
  font-family: SF Pro Display Bold;
  font-size: 16px;
  line-height: 24px;
  text-align: left;
  padding-top: 0px;
  padding-bottom: 0px;
  padding-left: 0px;
  padding-right: 0px;
  margin-top: 0px;
  margin-bottom: 0px;
  margin-left: 0px;
  margin-right: 0px;
}

<p
  class="c0"
  color="#000000"
>
  Test text component
</p>
`;
